components/tidb_query_expr/src/lib.rs

```diff
@@ -274,9 +274,9 @@ fn map_lower_sig(value: ScalarFuncSig, children: &[Expr]) -> Result<RpnFnMeta> {
         ));
     }
     if children[0].get_field_type().is_binary_string_like() {
-        Ok(lower_binary_fn_meta())
-    } else {
         Ok(lower_fn_meta())
+    } else {
+        Ok(lower_utf8_fn_meta())
```

Comment:
We don't need this check.

Reply:
We don't need check for `is_binary_string_like`. If `lower` is pushed to TiKV, it is for UTF-8 ones.

Reply:
This check is needed. See my comment below.

Reply:
This check is needed. See my comment below.